subsys/ieee802154/nrf_802154_configurator.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <init.h>
+#include <logging/log.h>
+#include <sys/__assert.h>
+#include <devicetree.h>
+#include <nrf_802154.h>
+#include <nrf_802154_config.h>
+
+LOG_MODULE_REGISTER(nrf_802154_configure);
+
+static uint8_t lna_gain_get(void)
+{
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)
+	return CONFIG_MPSL_FEM_NRF21540_RX_GAIN_DB;
+
+#elif IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)
+#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
+#error Node with label 'nrf_radio_fem' not found in the devicetree.
+#endif
+	return DT_PROP(DT_NODELABEL(nrf_radio_fem), rx_gain_db);
+
+#else
+	return 0U;
+#endif
+}
+
+static void ccaed_threshold_configure(void)
+{
+	nrf_802154_cca_cfg_t cca_cfg;
+
+	nrf_802154_cca_cfg_get(&cca_cfg);
+
+	/* Overwrite thresholds accounting for RX track gain. */
+	cca_cfg.ed_threshold = NRF_802154_CCA_ED_THRESHOLD_DEFAULT + lna_gain_get();
+	cca_cfg.corr_threshold = NRF_802154_CCA_CORR_THRESHOLD_DEFAULT + lna_gain_get();
+
+	nrf_802154_cca_cfg_set(&cca_cfg);
+}
+
+static int nrf_802154_configure(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	ccaed_threshold_configure();
+
+	return 0;
+}
+
+BUILD_ASSERT(CONFIG_KERNEL_INIT_PRIORITY_DEVICE < CONFIG_NRF_802154_RADIO_CONFIG_PRIO,
```

Comment:
I don't understand why do you compare with `CONFIG_KERNEL_INIT_PRIORITY_DEVICE `? The radio driver will get initialized at `POST_KERNEL`/`CONFIG_IEEE802154_NRF5_INIT_PRIO`, I think you should rather compare with that one, or simply re-use the upstream `CONFIG_IEEE802154_NRF5_INIT_PRIO` symbol and add +1 for later initialization.

Reply:
I would have happily add +1 if this had been permitted. So indeed I mixed up initialization priorities. Now the comparison is done against `CONFIG_IEEE802154_NRF5_INIT_PRIO`.

Reply:
Indeed, +1 would not work looking at the SYS_INIT API description, I wasn't aware of that, thanks for clarifying.